mllib/src/main/scala/org/apache/spark/ml/tree/impl/GradientBoostedTrees.scala

```diff
@@ -329,8 +330,16 @@ private[spark] object GradientBoostedTrees extends Logging {
       //       However, the behavior should be reasonable, though not optimal.
       baseLearnerWeights(m) = learningRate
 
+      if (pre_predError.getStorageLevel != StorageLevel.NONE ){
```

Comment:
First, I hardcoded MEMORY_AND_DISK  because it's just the same as the DecisionTree.train method internal do to some RDD, and i agree with that  because MEMORY_AND_DISK  is very compatible. MEMORY_AND_DISK  will cache RDD to memory only if memory is enough ,else will cache some to disk.  If you let users to set storeage level  they may set MEMORY_ONLY  which may cause serious problem  that  i met once .When the input data is very large and I set MEMORY_ONLY  , the training time is  almost 45 hours  most of which is wasted in  waiting for the memory releasing!

Second,  please read my code carefully because i 've unpersist previousPredError  after the loop.

Finally , this really relates to running out of memory because my experiment  without my improvement  occured  a lot of task failures  like : 
`(ExecutorLostFailure (executor 6 exited caused by one of the running tasks) Reason: Container killed by YARN for exceeding memory limits. 10.2 GB of 10 GB physical memory used. Consider boosting spark.yarn.executor.memoryOverhead.)`

I think this may be caused by : if you do not  cache previous predError RDD in the loop, then when it updates predError RDD every time , it will recaculate it from the head of the lineage ,  and after several rounds  when the lineage becomes long  the recaculation of all the previous predError RDD  may cost quite a few time and may also produce a  lot of tempory memory using . That may caused the task failures i met.

Can we get more people to check this issue?